task/task.go

```diff
@@ -0,0 +1,151 @@
+package task
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/satori/go.uuid"
+)
+
+type Action func() error
+
+type TaskObject struct {
+	ID string
+
+	ActiveStatus ActiveStatus
+	FinalStatus  FinalStatus
+}
+
+type Task interface {
```

Comment:
How about calling this `Service` ? Thus it's `task.Service` and would give a rather clear idea of what it is about. the controller could call it `TaskService` or `Tasks`